clang/test/CodeGen/cheri/cheri-implicit-casts.c

```diff
@@ -0,0 +1,129 @@
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
+// RUN: %cheri_cc1 -Wno-cheri -Wno-cheri-pointer-conversion -o - -disable-O0-optnone -emit-llvm %s \
+// RUN:   | opt -S -mem2reg | FileCheck %s --check-prefix=HYBRID
+// RUN: %cheri_cc1 -x c++ -Wno-cheri -Wno-cheri-pointer-conversion -o - -disable-O0-optnone -emit-llvm %s \
+// RUN:   | opt -S -mem2reg | FileCheck %s --check-prefix=HYBRID
+// RUN: %cheri_purecap_cc1 -Wno-cheri -Wno-cheri-pointer-conversion -o - -disable-O0-optnone -emit-llvm %s \
+// RUN:   | opt -S -mem2reg | FileCheck %s --check-prefix=PURECAP
+// RUN: %cheri_purecap_cc1 -x c++ -Wno-cheri -Wno-cheri-pointer-conversion -o - -disable-O0-optnone -emit-llvm %s \
+// RUN:   | opt -S -mem2reg | FileCheck %s --check-prefix=PURECAP
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// HYBRID-LABEL: @cap_to_long(
+// HYBRID-NEXT:  entry:
+// HYBRID-NEXT:    [[TMP0:%.*]] = call i8 addrspace(200)* @llvm.cheri.ddc.get()
+// HYBRID-NEXT:    [[TMP1:%.*]] = call i64 @llvm.cheri.cap.to.pointer.i64(i8 addrspace(200)* [[TMP0]], i8 addrspace(200)* [[C:%.*]])
+// HYBRID-NEXT:    ret i64 [[TMP1]]
+//
+// PURECAP-LABEL: @cap_to_long(
+// PURECAP-NEXT:  entry:
+// PURECAP-NEXT:    [[TMP0:%.*]] = call i64 @llvm.cheri.cap.address.get.i64(i8 addrspace(200)* [[C:%.*]])
```

Comment:
I know offset mode is dead, but we could also generate the purecap output for offset mode here to show that clang also uses address.get in offset mode. 
Hopefully adding
```
// RUN: %cheri_purecap_cc1 -x c++ -Wno-cheri -Wno-cheri-pointer-conversion -o - -disable-O0-optnone -emit-llvm %s \// RUN:   | opt -S -mem2reg | FileCheck %s --check-prefixes=PURECAP,PURECAP-OFFSET
```
will reuse almost all check lines and only add the few that differ.

Reply:
Not bothering as discussed on Slack